src/NBitcoin/Script.cs

```diff
@@ -409,14 +409,14 @@ public Script(IEnumerable<byte> data)
             this._Script = data.ToArray();
         }
 
-        public Script(byte[] data, bool compressed)
+        public Script(Network network, byte[] data, bool compressed)
```

Comment:
Script will never get any of the types that are in ConsensusFactory so we dont need to pass network in this case)

Use the override without network when calling `compressor.ReadWrite`